### ISSUE TITLE
Compile Error in up-to-date arch Linux

### ISSUE THREAD

**Issue by mgrewe** [zarquon42b/Morpho#11]
Hi Stephan,

I wanted to recompile Morpho for R version 3.5.1 (2018-07-02), using a quite recent glib (2.28-4) in up-to-date arch Linux. I got the following error during install.packages:

``` 
[...]
installing to /[...]/R/x86_64-pc-linux-gnu-library/3.5/Morpho/libs
** R
** data
** inst
** byte-compile and prepare package for lazy loading
terminate called after throwing an instance of 'std::runtime_error'
  what():  Mutex creation failed
/usr/lib64/R/bin/INSTALL: line 34:  2553 Done                    echo 'tools:::.install_packages()'
      2554 Aborted                 (core dumped) | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
Warning in install.packages :
  installation of package ‘Morpho’ had non-zero exit status
```

If you can point me to some potential code locations, I can try to debug the issue myself.

**Comment by zarquon42b** [comment]
Hi Martin,
did you use the cran version or the github version?
How did you try to install?
This seems to be an issue in the system as the tests (with all versions) seem to be fine: https://cran.r-project.org/web/checks/check_results_Morpho.html
Maybe try to install the tarball directly via R CMD INSTALL path_to_morpho.tar.gz

**Comment by mgrewe** [comment]
Thanks for the quick reply. Originally, I used build in `install.packages("Morpho")` and thus the latest package available via cran. I now tried compiling using Master from github via `R CMD INSTALL path_to_morpho.tar.gz` as you mentioned, but it gave me the same error. I assume, the build tests might not check against different glib versions and I am pretty sure, that the first occurrence of that error coincides with the glib update.
